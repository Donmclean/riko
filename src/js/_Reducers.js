import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import postsSelectorReducer from './components/Shared/PostSelector/postSelectorReducer';
import counterReducer from './components/Shared/CounterClicker/counterReducer';

export default combineReducers({
    postsSelectorReducer,
    counterReducer,
    routing: routerReducer
});