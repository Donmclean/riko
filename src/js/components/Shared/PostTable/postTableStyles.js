import styled from 'styled-components';

export const PostTableWrapper = styled.div``;

export const Table = styled.table`
    width: 100%;
    margin: auto;
    text-align: center;
`;

export const TableID = styled.td`
    width: 10%;
`;

export const TableTitle = styled.td`
    width: 30%;
`;

export const TableBody = styled.td`
    width: 60%;
`;