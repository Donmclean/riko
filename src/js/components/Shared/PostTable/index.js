import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

//Import Styled Components
import { PostTableWrapper, Table, TableID, TableTitle, TableBody } from './postTableStyles';

class PostsTable extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        //Notice how state is destructured and applied
        const { state } = this.props;
        return (
            <PostTableWrapper>
                <h1>Async Reducer State (Posts)</h1>
                <Table>
                    <thead>
                        <tr>
                            <TableID>ID</TableID>
                            <TableTitle>TITLE</TableTitle>
                            <TableBody>BODY</TableBody>
                        </tr>
                    </thead>
                    <tbody>
                        <tr>
                            <TableID>{state.postSelectorReducer.get('id')}</TableID>
                            <TableTitle>{state.postSelectorReducer.get('title')}</TableTitle>
                            <TableBody>{state.postSelectorReducer.get('body')}</TableBody>
                        </tr>
                    </tbody>
                </Table>
            </PostTableWrapper>
        );
    }
}

PostsTable.propTypes = {
    state: PropTypes.object.isRequired
};

//'state' gets applied to components' props here via 'connect'.
const mapStateToProps = (state) => ({state});

export default connect(mapStateToProps)(PostsTable);