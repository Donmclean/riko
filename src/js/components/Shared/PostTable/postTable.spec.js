import React from 'react';

//For testing components
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
const middlewares = []; //eg: [thunk] for async | eg: [epicMiddleware] for testing epics
const mockStore = configureStore(middlewares);

//This contains all reducers and their 'initialState' ready to be placed in 'mockStore' for testing
//This should be ideal for testing dummy components
import { mockStoreInitialState } from '../../../_Reducers';

//Component to be Tested
import PostTable from './index';

describe('PostTable Test Suite', () => {
    let store;

    beforeEach(() => {
        //Initialize mockStore with initial state of the reducers
        store = mockStore(mockStoreInitialState);
    });

    describe('PostTable Component', () => {
        it('renders <PostTable /> component successfully in to the DOM with required props & state', () => {

            //Verify props
            const wrapper = shallow(<PostTable store={store}/>);
            expect(wrapper.prop('store')).toEqual(store);

            //Verify that initial State of the store is rendered correctly
            expect(wrapper.state().storeState).toEqual(store.getState());

            //Render Component Tree
            const component = renderer.create(
                <PostTable store={store}/>
            );

            //Verify Component Snapshot
            //call component to JSON before matching snapshot
            expect(component.toJSON()).toMatchSnapshot();
        });
    });
});