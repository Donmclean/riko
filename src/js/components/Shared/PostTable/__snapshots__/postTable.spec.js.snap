exports[`PostTable Test Suite PostTable Component renders <PostTable /> component successfully in to the DOM with required props & state 1`] = `
<div
  className="a">
  <h1>
    Async Reducer State (Posts)
  </h1>
  <table
    className="iprvVv">
    <thead>
      <tr>
        <td
          className="bBeomu">
          ID
        </td>
        <td
          className="ieFpxY">
          TITLE
        </td>
        <td
          className="jZgYSW">
          BODY
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="bBeomu">
          0
        </td>
        <td
          className="ieFpxY">
          Initial State Title
        </td>
        <td
          className="jZgYSW">
          Pending Request..
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
